### ISSUE TITLE
Create a "All Build Items" page

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#9945]
**Description**
When working on a Quarkus extension, I often wonder myself "how can I do this ?" or "is there a build item to do this?".

Quarkus build mechanism offers a lot of pre-defined functonalities thanks to the numerous build items. Even if we can discover them by searching them inside our IDE, it will be more convenient to have a single page that list them all with some documentation.

**Implementation ideas**
Inspired by what has been done for listing all configuration properties we can generate a "All Build Items" page, as this is a page for extension author it didn't need to be very user friendly and can only be linked from the extension author guide.

This page can list all build items by name and copy the JavaDoc inside the page as documentation. Grouping them by extension and having a search field would be great.

**Comment by ppalaga** [comment]
I was just about to file the same issue. A catalog of build items would be very handy for everybody working on Quarkus extensions.

The bottom line is that the JavaDoc of existing BuildItems is often missing or does not contain all relevant information. E.g. the JavaDOc of `AdditionalApplicationArchiveBuildItem` says

> An additional application archive. This build item can only be consumed, it should not be produced by build steps.

What I am missing there:

* Definition of "application archive" and "additional application archive".
* What are the situations in which I should/shouldn't use this BuildItem?
* A basic example

**Comment by gastaldi** [comment]
@ppalaga Here is what I have accomplished so far: 

[build-items.html.zip](https://github.com/quarkusio/quarkus/files/5233752/build-items.html.zip)


I'll share the sources I used to generate it afterwards

**Comment by gastaldi** [comment]
Run the following command in the root of a cloned Quarkus Core repository to see the generated Asciidoc: 
```
find extensions -name "*BuildItem.java" | jbang builditemdoc@quarkusio > docs/src/main/asciidoc/build-items.adoc
```

**Comment by gastaldi** [comment]
Here is the JBang script that generates it: https://github.com/quarkusio/jbang-catalog/blob/master/quarkusbuilditemdoc.java

**Comment by ppalaga** [comment]
Great work, thanks @gastaldi !

I am missing the buildItems from the `quarkus-core` extension in the attached html file. Those happen to be the most important ones 8-)

**Comment by gastaldi** [comment]
> I am missing the buildItems from the `quarkus-core` extension in the attached html file. Those happen to be the most important ones 8-)

Ah yes, good catch. Please find the updated version in https://github.com/quarkusio/quarkus/pull/12147

**Comment by loicmathieu** [comment]
Yeah, this is cool !
I'll comment in the PR.

**Comment by ppalaga** [comment]
@gastaldi is https://quarkus.io/guides/all-builditems the URL if the new BuildItems page?

**Comment by gastaldi** [comment]
@ppalaga yes, there should be a link in https://quarkus.io/guides/building-my-first-extension to there
